src/client/api/restful/get_outputs_id.c

```diff
@@ -6,6 +6,172 @@
+    if (next->query_item) {
+      LL_APPEND(*list, next);
+      return 0;
+    } else {
+      free(next);
+    }
+  }
+  return -1;
+}
+
+size_t get_outputs_query_str_len(outputs_query_list_t *list) {
+  size_t query_str_len = 0;
+  outputs_query_list_t *elm;
+  LL_FOREACH(list, elm) {
+    switch (elm->query_item->type) {
+      case QUERY_PARAM_ADDRESS:
+        query_str_len += strlen(OUTPUTS_QUERY_ADDRESS_KEY);
+        query_str_len += strlen(elm->query_item->param);
+        query_str_len += 2;  // For "&" params seperator and "=" params assignment
+        break;
+      case QUERY_PARAM_DUST_RET:
+        query_str_len += strlen(OUTPUTS_QUERY_DUST_RET_KEY);
+        query_str_len += strlen(elm->query_item->param);
+        query_str_len += 2;  // For "&" params seperator and "=" params assignment
+        break;
+      case QUERY_PARAM_DUST_RET_ADDR:
+        query_str_len += strlen(OUTPUTS_QUERY_DUST_RET_ADDR_KEY);
+        query_str_len += strlen(elm->query_item->param);
+        query_str_len += 2;  // For "&" params seperator and "=" params assignment
+        break;
+      case QUERY_PARAM_SENDER:
+        query_str_len += strlen(OUTPUTS_QUERY_SENDER_KEY);
+        query_str_len += strlen(elm->query_item->param);
+        query_str_len += 2;  // For "&" params seperator and "=" params assignment
+        break;
+      case QUERY_PARAM_TAG:
+        query_str_len += strlen(OUTPUTS_QUERY_TAG_KEY);
+        query_str_len += strlen(elm->query_item->param);
+        query_str_len += 2;  // For "&" params seperator and "=" params assignment
+        break;
+      case QUERY_PARAM_PAGE_SIZE:
+        query_str_len += strlen(OUTPUTS_QUERY_PAGE_SIZE_KEY);
+        query_str_len += strlen(elm->query_item->param);
+        query_str_len += 2;  // For "&" params seperator and "=" params assignment
+        break;
+      case QUERY_PARAM_CURSOR:
+        query_str_len += strlen(OUTPUTS_QUERY_CURSOR_KEY);
+        query_str_len += strlen(elm->query_item->param);
+        query_str_len += 2;  // For "&" params seperator and "=" params assignment
+        break;
+      case QUERY_PARAM_STATE_CTRL:
+        query_str_len += strlen(OUTPUTS_QUERY_STATE_CTRL_KEY);
+        query_str_len += strlen(elm->query_item->param);
+        query_str_len += 2;  // For "&" params seperator and "=" params assignment
+        break;
+      case QUERY_PARAM_GOV:
+        query_str_len += strlen(OUTPUTS_QUERY_GOV_KEY);
+        query_str_len += strlen(elm->query_item->param);
+        query_str_len += 2;  // For "&" params seperator and "=" params assignment
+        break;
+      case QUERY_PARAM_ISSUER:
+        query_str_len += strlen(OUTPUTS_QUERY_ISSUER_KEY);
+        query_str_len += strlen(elm->query_item->param);
+        query_str_len += 2;  // For "&" params seperator and "=" params assignment
+        break;
+      default:
+        break;
+    }
+  }
+  query_str_len--;  // Remove the "&" params seperator at the end
```

Comment:
What if there is no query in a list?

Reply:
if (query_str_len > 0) {
    query_str_len--;  // Remove the "&" params seperator at the end
  }